src/utils/index.ts

```diff
@@ -283,3 +285,28 @@ export function shallowEqual(objA: OAny, objB: OAny): boolean {
 
   return true
 }
+
+export function linkify(text: string): string {
+  const escaped = escapeHtml(text);
+  const linked = autoLink(escaped);
+  return linked;
+}
+
+const linkPattern = /(^|[\s\n]|<[A-Za-z]*\/?>)((?:(?:https?):\/\/|lichess\.org\/)[\-A-Z0-9+\u0026\u2019@#\/%?=()~_|!:,.;]*[\-A-Z0-9+\u0026@#\/%=~()_|])/gi;
+
+function linkReplace(_: string, before: string, url: string) {
+  const fullUrl = url.indexOf('http') === 0 ? url : 'https://' + url;
+  const minUrl = url.replace(/^(?:https:\/\/)?(.+)$/, '$1');
+  return before + '<a href="#" onClick="window.open(\'' + fullUrl + '\', \'_system\')">' + minUrl + "</a>";
```

Comment:
String concatenation is hard to understand here. Please rewrite it so:
`<a href="#" onClick="window.open('${fullUrl}', '_blank')">${minUrl}</a>`

Notice the `_blank` instead of system: it opens a link in the in-app browser. This is consistent with other links in the app.

Reply:
The issue with `_blank` is that it won't redirect the user to the appropriate section of the app if the app could handle the link. For example, if the user clicked on a link like `https://lichess.org/tournament/QecWLuj9` and we set the target to `_blank` it will open that website version of the tournament page in an inapp browser. By using the `_system` target, the user will be prompted to either open the link in a proper browser or the app. If they choose the app, it will fall through out normal deep link regex rules defined in `config.xml` and `deepLinks.ts` so if the app can handle the link the user is directed to the correct app component and if the app can't handle the link, the link is opened in the inapp browser.
This is all consistent with how clicking a lichess.org link outside the app would work. While it involves kicking the control flow out to the OS for a moment, I think it's the best approach as it doesn't involve reimplementing the rules we've defined in `config.xml` again.